features/authentication.feature

```diff
@@ -332,3 +332,15 @@ Feature: OAuth authentication
     Then the output should contain "github.com username:"
     And the output should contain "missing user"
     And the file "../home/.config/hub" should not contain "user"
+    
+  Scenario: Config file is not writeable, should exit before asking for credentails
+      Given I set the environment variables to:
+      | variable           | value      |
+      | HUB_CONFIG | /InvalidConfigFile |
```

Comment:
In addition to testing a custom location via HUB_CONFIG environment variable, can you make a scenario where hub tries to write the config in the default location, but that location isn't writeable or an existing file is read only?

We could maybe use these built-in Aruba steps:

```rb
Given /^a directory named "([^"]*)" with mode "([^"]*)"$/ do |dir_name, dir_mode|
  create_dir(dir_name)
  chmod(dir_mode, dir_name)
end

Given /^a file named "([^"]*)" with mode "([^"]*)" and with:$/ do |file_name, file_mode, file_content|
  write_file(file_name, file_content)
  chmod(file_mode, file_name)
end
```

The default config location will be `$HOME/.config/hub`, where "HOME" in the context of tests is [defined here](https://github.com/github/hub/blob/4a7f8cb53a4ef1364e348d45b45628430c077eba/features/support/env.rb#L38).